Sources/Datadog/RUMMonitor.swift

```diff
@@ -148,9 +148,14 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
 
     /// RUM Attributes shared with other Feature registered in core.
     internal struct Attributes {
+        /// The ID of RUM application (`String`).
         internal static let applicationID = "application_id"
+        /// The ID of current RUM session (standard UUID `String`, lowercased).
+        /// In case the session is rejected (not sampled), RUM context is set to empty (`[:]`) in core.
```

Comment:
I'm not sure how this comment relates to this property? Is this session_id set to '[:]' if it was not sampled?

Reply:
This is interesting problem 👍. You're right and this comment doesn't directly relate to `sessionID` field. It relates to the behaviour that all `applicationID`, `sessionID`, `viewID` and `userActionID` have in common: if session is not sampled, none of these will appear in RUM Feature's context notified by core.

In other words, if session is sampled, core context will be this:
```swift
// ...
featureAttributes: [
   "rum": [
      Attributes.applicationID: "<...>",
      Attributes. sessionID: "<...>",
      Attributes.viewID: "<...>",
      Attributes.userActionID: "<...>",
   ]
]
```
and if it was rejected (not sampled), this:
```
// ...
featureAttributes: [
   "rum": [:]
]
```

Now, because these attributes describe how RUM context will be provided to other Features (Logs, Traces, Session Replay, ...) - it's IMO quite critical to have this behaviour well described and documented. I agree that single `Attributes.sessionID` is not the best place, but it's the only one that I found so far.

Maybe what we need is `RUMContextAttributes` instead of `Attributes` and having this behaviour described at the `struct` level. WDYT @fuzzybinary ?